subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -32,32 +33,46 @@
 import org.gradle.internal.logging.events.StyledTextOutputEvent;
 import org.gradle.internal.logging.text.StyledTextOutput;
 import org.gradle.internal.progress.BuildOperationCategory;
+import org.gradle.internal.time.TimeProvider;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
 
 /**
  * An {@code org.gradle.logging.internal.OutputEventListener} implementation which generates output events to log the
  * progress of operations.
  */
 public class GroupingProgressLogEventGenerator extends BatchOutputEventListener {
     static final String EOL = SystemProperties.getInstance().getLineSeparator();
+    static final long LONG_RUNNING_TASK_OUTPUT_FLUSH_TIMEOUT = TimeUnit.SECONDS.toMillis(5);
```

Comment:
It's already in milliseconds. I used `TimeUnit` so the scale is clear. We can add it as a constructor argument. During the test, I use a mock `TimeProvider` to speed up the test.